### PR TITLE
Halt on closed user output

### PR DESCRIPTION
**Pull request by GavinMendelGleason**

### CODE REVIEW

**Review by spl — DISMISSED** [review]
How can we have an automated test for this?

File: src/cli/main.pl [hunk 1]
```diff
@@ -46,7 +46,9 @@
             halt(0)
         ),
         Exception,
-        (   Exception = error(Error,context(prolog_stack(Stack),_)),
+        (   Exception = error(io_error(write,user_output),_)
+        ->  halt(0)
```
[the remark this answers is not in the document]
Comment (GavinMendelGleason) [reply src/cli/main.pl:50]: I'm not sure - what do unix utils do on truncation of the output stream?

[the remark this answers is not in the document]
Comment (GavinMendelGleason) [reply src/cli/main.pl:50]: ls appears to exit with 0.

**Comment by matko** [comment]
Fixes #1148.

**Comment by GavinMendelGleason** [comment]
> How can we have an automated test for this?

I don't think a test adds much value since it is unlikely to suffer regression. The solution is a hack, and whatever bugs exist with it will bite us anyhow.

**Review by spl — APPROVED** [review]

**Pull request merged**